### PR TITLE
VMSS: exposing the `requireGuestProvisionSignal` field for `2019-03-01`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

---

Noticed this field was being sent across the wire but isn't in the Swagger:

```
{
	"name": "tom-dev3",
	"id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/tom-vmss-linux-dev/providers/Microsoft.Compute/virtualMachineScaleSets/tom-dev3",
	"type": "Microsoft.Compute/virtualMachineScaleSets",
	"location": "westeurope",
	"tags": {},
	"sku": {
		"name": "Standard_F2",
		"tier": "Standard",
		"capacity": 3
	},
	"properties": {
		"singlePlacementGroup": false,
		"upgradePolicy": {
			"mode": "Manual"
		},
		"virtualMachineProfile": {
			"osProfile": {
				"computerNamePrefix": "tomdev",
				"adminUsername": "tharvey",
				"linuxConfiguration": {
					"disablePasswordAuthentication": true,
					"ssh": {
						"publicKeys": [{
							"path": "/home/tharvey/.ssh/authorized_keys",
							"keyData": "XXXXX"
						}]
					},
					"provisionVMAgent": true
				},
				"secrets": [],
				"allowExtensionOperations": true,
				"requireGuestProvisionSignal": true
			},
			"storageProfile": {
				"osDisk": {
					"createOption": "FromImage",
					"caching": "ReadWrite",
					"writeAcceleratorEnabled": false,
					"managedDisk": {
						"storageAccountType": "Standard_LRS"
					}
				},
				"imageReference": {
					"publisher": "Canonical",
					"offer": "UbuntuServer",
					"sku": "16.04-LTS",
					"version": "latest"
				}
			},
			"networkProfile": {
				"networkInterfaceConfigurations": [{
					"name": "example",
					"properties": {
						"primary": true,
						"enableAcceleratedNetworking": false,
						"dnsSettings": {
							"dnsServers": []
						},
						"enableIPForwarding": false,
						"ipConfigurations": [{
							"name": "internal",
							"properties": {
								"primary": true,
								"subnet": {
									"id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/tom-vmss-linux-dev/providers/Microsoft.Network/virtualNetworks/tom-vmss-linux-vnet/subnets/internal"
								},
								"privateIPAddressVersion": "IPv4"
							}
						}]
					}
				}]
			},
			"priority": "Regular"
		},
		"provisioningState": "Failed",
		"overprovision": false,
		"doNotRunExtensionsOnOverprovisionedVMs": false,
		"uniqueId": "24e91467-72b2-4cd8-94b9-c225e363692a",
		"platformFaultDomainCount": 0
	}
}
```

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json [hunk 1]
```diff
@@ -6935,7 +6935,11 @@
         "allowExtensionOperations": {
           "type": "boolean",
           "description": "Specifies whether extension operations should be allowed on the virtual machine. <br><br>This may only be set to False when no extensions are present on the virtual machine."
-        }
+        },
```
Comment (yungezz) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: adding new property to existing version is breaking change. suggest to add in new service tag. definition of breaking change https://github.com/Azure/adx-documentation-pr/wiki/Breaking-changes-guidelines

Comment (tombuildsstuff) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: @yungezz this is coming across the wire already (see the response above) and thus presumably defaulted - so this is just missing from the Swagger definition?

Comment (tombuildsstuff) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: taking a look into this it appears that scaleInPolicy is supported in the API but missing from the Swagger too: https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/createorupdate

Comment (tombuildsstuff) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: also it appears the repository being linked to isn't public?

**Comment by yungezz** [comment]
hi @tombuildsstuff could you pls fix CI failure then the PR could be merged?
Here is error reference for Semantic and ModelValidation: https://github.com/Azure/adx-documentation-pr/wiki/Semantic-and-Model-Violations-Reference

**Comment by tombuildsstuff** [comment]
@yungezz unfortunately that link returns a 404:

Screenshot 2019-09-25 at 10 03 40

**Comment by yungezz** [comment]
@tombuildsstuff don't worry, we're trying to find a public place for the doc.

**Comment by tombuildsstuff** [comment]
@yungezz all good - I've opened https://github.com/Azure/azure-rest-api-specs/issues/7315 to see if that can be OSS'd

In the meantime, what's the best path to proceed with this?

**Comment by yungezz** [comment]
offline version for reference.
[Semantic and Model Violations Reference.zip](https://github.com/Azure/azure-rest-api-specs/files/3651747/Semantic.and.Model.Violations.Reference.zip)

**Comment by yungezz** [comment]
hi @tombuildsstuff any update on the PR?

**Comment by yungezz** [comment]
HI @tombuildsstuff could you pls fix CI failure?

**Review by yungezz — APPROVED** [review]

**Pull request merged**